docs/reference/composition.md

```diff
@@ -405,6 +405,34 @@ field.
   toFieldPath: status.zone
 ```
 
+Both `FromCompositeFieldPath` and `ToCompositeFieldPath` can take a wildcarded
+field path containing arrays and patch each array element on a specified
+field.
```

Comment:
Looks like this is not correct. `fromFieldPath` cannot take a wildcard field path as you also stated in https://github.com/crossplane/crossplane/pull/2791#issuecomment-1127853801

UX-wise, we should try supporting this, but I don't think it makes sense to have wildcards in `fromFieldPath` in practice. 

In addition to documenting this properly, I would also suggest adding a check that throws an error if `fromFieldPath` has a wildcard in it since this could be quite confusing.

Reply:
Yep this language is totally wrong. Fixed.

Reply:
Ah, I figured out why it didn't seem to make any sense. Wildcards _do_ work for both `FromCompositeFieldPath` and `ToCompositeFieldPath` type patches, but _only_ for the `toFieldPath` parameter in either type. (Revisiting months old work is difficult when trying to remember what you were doing). I'll clarify the language to make it clear that wildcards in `toFieldPath` are the only valid input in both of these cases